pkg/provider/passworddepot/passworddepot.go

```diff
@@ -0,0 +1,189 @@
+type PasswordDepot struct {
+	client   Client
+	database string
+}
+
+func (p *PasswordDepot) ValidateStore(store esv1beta1.GenericStore) error {
+	return nil
+}
+
+func (a *PasswordDepot) Capabilities() esv1beta1.SecretStoreCapabilities {
+	return esv1beta1.SecretStoreReadOnly
+}
+
+// Client for interacting with kubernetes cluster...?
+type passwordDepotClient struct {
+	kube      kclient.Client
+	store     *esv1beta1.PasswordDepotProvider
+	namespace string
+	storeKind string
+}
+type Provider struct{}
+
+// Set gClient credentials to Access Token.
+func (c *passwordDepotClient) getAuth(ctx context.Context) (string, string, error) {
+	credentialsSecret := &corev1.Secret{}
+	credentialsSecretName := c.store.Auth.SecretRef.Credentials.Name
+	if credentialsSecretName == "" {
+		return "", "", fmt.Errorf(errPasswordDepotCredSecretName)
+	}
+	objectKey := types.NamespacedName{
+		Name:      credentialsSecretName,
+		Namespace: c.namespace,
+	}
+	// only ClusterStore is allowed to set namespace (and then it's required)
+	if c.storeKind == esv1beta1.ClusterSecretStoreKind {
+		if c.store.Auth.SecretRef.Credentials.Namespace == nil {
+			return "", "", fmt.Errorf(errInvalidClusterStoreMissingSAKNamespace)
+		}
+		objectKey.Namespace = *c.store.Auth.SecretRef.Credentials.Namespace
+	}
+
+	err := c.kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		return "", "", fmt.Errorf(errFetchSAKSecret, err)
+	}
+
+	username := credentialsSecret.Data["username"]
+	password := credentialsSecret.Data["password"]
+	if (username == nil) || (len(username) == 0 || password == nil) || (len(password) == 0) {
+		return "", "", fmt.Errorf(errMissingSAK)
+	}
+
+	return string(username), string(password), nil
+}
+
+// Function newPasswordDepotProvider returns a reference to a new instance of a 'PasswordDepot' struct.
+func NewPasswordDepotProvider() *PasswordDepot {
+	return &PasswordDepot{}
+}
+
+// Method on PasswordDepot Provider to set up client with credentials and populate projectID.
+func (g *PasswordDepot) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	storeSpec := store.GetSpec()
+	if storeSpec == nil || storeSpec.Provider == nil || storeSpec.Provider.PasswordDepot == nil {
+		return nil, fmt.Errorf("no store type or wrong store type")
+	}
+	storeSpecPasswordDepot := storeSpec.Provider.PasswordDepot
+
+	// TODO check if all members needed
+	cliStore := passwordDepotClient{
+		kube:      kube,
+		store:     storeSpecPasswordDepot,
+		namespace: namespace,
+		storeKind: store.GetObjectKind().GroupVersionKind().Kind,
+	}
+
+	username, password, err := cliStore.getAuth(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new PasswordDepot client using credentials and options
+	passworddepotClient, err := NewPasswortDepotApi(storeSpecPasswordDepot.Host, username, password, "8714", true)
```

Comment:
I think by default insecure skip verify should absolutely be false. I think there are also sec checks that should fail if it's being used. So definitely do not do that.

Rather offer the ability to pass in certitifactes.

Reply:
Removed insecure skip verify.